### PR TITLE
Refactor BSD build support to be extendable and add NetBSD

### PR DESCRIPTION
**Pull request by vazub**
Allow building `wren_cli` for both FreeBSD and NetBSD, and allow trivial future extension for other BSDs, if needed.

Unfortunately, Premake bundles all separate BSDs under a single `system:bsd` definition, making it impossible to set up build conditionals purely with `--os=` option, and therefore adding additional option was required to circumvent this problem. 

Moreover, it made sense to define dedicated `gmake.<*bsd>` folder location for each system's project files, to have clear separation of concerns, and since BSD `make` is not usable in our case, folder prefix was deliberately changed to `gmake.` to provide additional hint for those building the project from source, regarding what `make` variant they are required to use.

### CODE REVIEW

File: projects/premake/premake5.lua [hunk 1]
```diff
@@ -1,164 +1,182 @@
...
-      "../../deps/libuv/src/unix/posix-hrtime.c",
-      "../../deps/libuv/src/unix/random-getrandom.c",
-    }
+workspace "wren-cli"
+  configurations { "Release", "Debug" }
+  platforms { "64bit", "32bit", "64bit-no-nan-tagging" }
+  defaultplatform "64bit"
+  location ("../" .. _ACTION)
+
+  newoption {
```
Comment (vazub) [inline projects/premake/premake5.lua:7]: Added this

**Comment by ruby0x1** [comment]
Thanks @vazub looks like a reasonable approach. I'll maybe rearrange a tiny bit but otherwise will test/review properly soon.

**Comment by vazub** [comment]
Looks like there is no interest in this. Closing.

**Comment by ruby0x1** [comment]
The issue isn't really interest, mostly time, and the burden of testing and maintaining those platforms for each release is a consideration. 

I think this is a reasonable approach as mentioned, though there's a file that has it's entire contents marked as a diff which we should probably correct before this can be merged either way.

File: projects/premake/premake5.lua [hunk 2]
```diff
@@ -1,164 +1,182 @@
-workspace "wren-cli"
```
Comment (ruby0x1) [inline projects/premake/premake5.lua:1]: Let's fix up the line endings or whatever has caused this entire file to be a diff :)

**Comment by vazub** [comment]
@ruby0x1 kind request to address my question in the resolved section above, because I am not sure how to proceed.

**Pull request closed without merging**